mono/metadata/debug-mono-ppdb.c

```diff
@@ -70,29 +81,47 @@ enum {
 };
 
 static gboolean
-get_pe_debug_guid (MonoImage *image, guint8 *out_guid, gint32 *out_age, gint32 *out_timestamp)
+get_pe_debug_info (MonoImage *image, guint8 *out_guid, gint32 *out_age, gint32 *out_timestamp, guint8 **ppdb_data,
+				   int *ppdb_uncompressed_size, int *ppdb_compressed_size)
 {
 	MonoPEDirEntry *debug_dir_entry;
 	ImageDebugDirectory *debug_dir;
+	int idx;
+	gboolean guid_found = FALSE;
+
+	*ppdb_data = NULL;
 
 	debug_dir_entry = &image->image_info->cli_header.datadir.pe_debug;
 	if (!debug_dir_entry->size)
 		return FALSE;
 
 	int offset = mono_cli_rva_image_map (image, debug_dir_entry->rva);
-	debug_dir = (ImageDebugDirectory*)(image->raw_data + offset);
-	if (debug_dir->type == 2 && debug_dir->major_version == 0x100 && debug_dir->minor_version == 0x504d) {
-		/* This is a 'CODEVIEW' debug directory */
-		CodeviewDebugDirectory *dir = (CodeviewDebugDirectory*)(image->raw_data + debug_dir->pointer);
-
-		if (dir->signature == 0x53445352) {
-			memcpy (out_guid, dir->guid, 16);
-			*out_age = dir->age;
-			*out_timestamp = debug_dir->time_date_stamp;
-			return TRUE;
+	for (idx = 0; idx < debug_dir_entry->size / sizeof (ImageDebugDirectory); ++idx) {
+		debug_dir = (ImageDebugDirectory*)(image->raw_data + offset) + idx;
+		if (debug_dir->type == DEBUG_DIR_ENTRY_CODEVIEW && debug_dir->major_version == 0x100 && debug_dir->minor_version == 0x504d) {
```

Comment:
Are these correct for big endian?

Reply:
It looks like it will fail on big endian but so will the old code so at least it doesn't regress?

@vargaz what needs to be done to land this?

Reply:
Accepts the costs of it.